internal/controller/pipelinerollout_controller.go

```diff
@@ -69,13 +68,20 @@ var (
 	initTime             time.Time
 )
 
-// PipelineSpec keep track of minimum number of fields we need to know about
+// PipelineSpec keeps track of minimum number of fields we need to know about
 type PipelineSpec struct {
 	InterStepBufferServiceName string    `json:"interStepBufferServiceName"`
-	Lifecycle                  lifecycle `json:"lifecycle,omitempty"`
+	Lifecycle                  Lifecycle `json:"lifecycle,omitempty"`
 }
 
-type lifecycle struct {
+func (pipeline PipelineSpec) getISBSvcName() string {
+	if pipeline.InterStepBufferServiceName == "" {
+		return "default"
+	}
+	return pipeline.InterStepBufferServiceName
+}
```

Comment:
This function existed previously but was not a function of `PipelineSpec` - thought this made more sense